### ISSUE TITLE
.NET Core designer uses LicenseManager.UsageMode = LicenseUsageMode.Runtime instead of LicenseUsageMode.Designtime

### ISSUE THREAD

**Issue by janseris** [dotnet/winforms#10960]
### Environment

Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.8.6

### .NET version

.NET 8

### Did this work in a previous version of Visual Studio and/or previous .NET release?

This works correctly in .NET Framework.

### Issue description

Designer does not respect LicenseManager.UsageMode to be LicenseUsageMode.Designtime.
This is very important so that code in constructor and OnLoad event handler can be selected which has to be skipped when designer executes the UserControl.
Code in OnLoad handler can probably be handled by Component.DesignMode (I have a note from older development that Component.DesignMode does not work in constructor.

Consequences are seen when you develop a UserControl which uses external dependencies in constructor body which are not available during design time and this UserControl is placed into a Form.

### Steps to reproduce

![obrazek](https://github.com/dotnet/winforms/assets/64279914/5d12ab64-4192-415c-97c9-d15c90c1d5c9)

**Comment by elachlan** [comment]
@Olina-Zhang do you know if this issue is already filed in the designer project?

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang do you know if this issue is already filed in the designer project?

Yes, we have a designer issue in private repo: https://github.com/microsoft/winforms-designer/issues/2640 - closed, there has more discussion about LicenseManager.UsageMode. 

@KlausLoeffelmann could you help look at this issue and give customer some suggestion if you have? 
FYI: @merriemcgaw

**Comment by janseris** [comment]
I found a workaround however it cost me approx. 3 hours to find out that this was the cause. One of the reasons is that code which is executed by designer cannot be debugged (that's not an issue, just saying).

`Assembly.GetEntryAssembly().GetName().Name == "DesignToolsServer"`

**Comment by merriemcgaw** [comment]
@janseris so glad you found a workaround! @KlausLoeffelmann I'd love your thoughts on the subject, and whether there is something we can do to make it easier - or perhaps get a docs change to reflect the new state of the world.

**Comment by janseris** [comment]
> @janseris so glad you found a workaround! @KlausLoeffelmann I'd love your thoughts on the subject, and whether there is something we can do to make it easier - or perhaps get a docs change to reflect the new state of the world.

Well, it should probably reflect .NET Framework behavior because more people might get confused?

**Comment by KlausLoeffelmann** [comment]
AFAIK, we are not supporting a LicenseManager-based concept in the Out-Of-Process Designer and also have no plans to do so.

**Comment by janseris** [comment]
> AFAIK, we are not supporting a LicenseManager-based concept in the Out-Of-Process Designer and also have no plans to do so. The reason `LicenceManager` is returning `Runtime` is most likely that there isn't any context which we actively maintain, and therefore the LicenseManager resorts to its default value.

Shouldn't there be a warning then? Breaking change
